ios/objective-c/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleApp/AppDelegate.m

```diff
@@ -0,0 +1,90 @@
+    //    '[Ampli instance]' is the default instance of Ampli
+    //
+    //    When you pull your tracking plan you can use the defaults and call load() without arguments
+    //    This requires connecting your account via `ampli pull` which will set you API key in the generated Ampli SDK
+    //
+    //    OR Specify a AmpliEnvironment
+    //    [ampli load:[LoadOptions initWithEnvironment:development]];
+    //
+    //    OR Provide a specific Amplitude API key
+    //    [ampli load:[LoadOptions initWithApiKey:@"Custom api key"]];
+    //
+    //    OR Use an existing Amplitude instance
+    //    requires `#import "Amplitude.h"`
+    //    Amplitude *instance = [Amplitude instanceWithName:@"instanceName"];
+    //    [instance initializeApiKey:@"Custom api key"];
+    //    [ampli load:[LoadOptions initWithAmplitudeInstance:instance]];
+    //
+    //    For testing you can disable ampli
+    //    [ampli load:[LoadOptions initWithDisabled:ENV.IS_TESTING ? YES : NO]];
+    //
+    //    Make as many Ampli instances as you want
+    //    Ampli ampli2 = [Ampli new];
+    //    [ampli2 load:[LoadOptions initWithApiKey:@"Custom api key"]];
+    NSString *apiKey = [[[NSProcessInfo processInfo] environment] objectForKey:@"AMPLITUDE_API_KEY"];
+    Ampli *ampli = [Ampli instance];
+    [ampli load:[LoadOptions builderBlock:^(LoadOptionsBuilder *builder) {
+        builder.apiKey =apiKey;
+    }]];
+    IdentifyProperties *identifyProperties = [IdentifyProperties new];
+    identifyProperties.optionalArray = [NSArray arrayWithObjects:@"optional string", nil];
+    identifyProperties.requiredNumber = @22;
+    [ampli identify:@"ampli-objc-user" properties:identifyProperties];
+    [ampli eventNoProperties];
+    NSMutableDictionary *extraDict = [NSMutableDictionary new];
+    [extraDict setObject:@"extra test"
+              forKey:@"test"];
+    EventMaxIntForTestProperties *eventMaxIntForTestProperties = [EventMaxIntForTestProperties new];
+    eventMaxIntForTestProperties.intMax10 = @20;
+    [ampli eventMaxIntForTest:eventMaxIntForTestProperties extra:extraDict];
+    [ampli eventWithConstTypes:extraDict];
+    EventWithAllPropertiesProperties *eventWithAllPropertiesProperties = [EventWithAllPropertiesProperties new];
+    eventWithAllPropertiesProperties.requiredArray = [NSArray arrayWithObjects:@"array element 1", @"array element 2", nil];
+    eventWithAllPropertiesProperties.requiredBoolean = @YES;
+    eventWithAllPropertiesProperties.requiredEnum = [RequiredEnum enum1];
+    eventWithAllPropertiesProperties.requiredInteger = @10;
+    eventWithAllPropertiesProperties.requiredNumber = @2.0;
+    eventWithAllPropertiesProperties.requiredString = @"required string";
+    EventWithAllProperties *eventWithAllProperties = [EventWithAllProperties initWithEventProperties:eventWithAllPropertiesProperties];
+    [ampli track:eventWithAllProperties];
```

Comment:
This usage is pretty verbose. Could we update the event classes to use a combination of a builder block and required arguments to 1) better enforce required/optional properties 2) make the usage less verbose.

In the [Itly ObjC v2 we had this usage](https://github.com/amplitude/itly-tools/blob/master/sdk-examples/ios-objc-v2/ItlyObjCV2App/AppDelegate.m#L103-L106) which I think would be nice here as well:
```
[ampli track:[EventWithEnumTypes requiredEnum:EventWithEnumTypesRequiredEnumRequiredEnum1
                                           builderBlock:^(EventWithEnumTypesBuilder * b) {
        b.optionalEnum = EventWithEnumTypesOptionalEnumOptionalEnum1;
    }]];
```

Reply:
@justin-fiedler  One concern here is adding builderBlock for Event is not really resolving the verbose in general, as we normally encourage to use the method `eventWithAllProperties` but not the track method right? And in that case it would still be verbose as it is now. If we want to support builder pattern, I guess supporting that in properties level would be better than supporting in event level. Maybe we can make this part of the improvements we plan for quicktype.